### PR TITLE
Add notification channels

### PR DESCRIPTION
**Pull request by ignatiusukwuoma**
#### What does this PR do?

Allow an organization to select their preferred channel for notifications.

#### Description of Task to be completed?

Add functionality that allows organization's administrator to select the preferred channel for notification upon actions on data requests to the organizations. The administrator should also be able to view the existing notification channels.

#### How should this be manually tested?

Start the server using `paster serve /etc/ckan/default/production.ini`
Log in as an organization owner/administrator
Click on the `Organisations` tab and select the specific organization then click on the `Manage` button
Click on the `Channels` tab. A list of existing notification channels will be displayed.
To add a new notification channel, click on the `Add Channel` button, select the preferred channel, fill in the required details then click on the `Get Notifications` button

#### What are the relevant issues?

Issue #1

#### Screenshots

screen shot 2017-10-17 at 12 33 06

screen shot 2017-10-17 at 12 33 22

screen shot 2017-10-17 at 12 35 18

screen shot 2017-10-17 at 12 35 46

### CODE REVIEW

File: ckanext/notify/actions.py [hunk 1]
```diff
@@ -0,0 +1,474 @@
...
+ValidationError = logic.ValidationError
+
+
+def _dictize_slack_details(slack_details):
+
+    # Convert the slack details into a dict
+    data_dict = {
+        'id': slack_details.id,
+        'webhook_url': slack_details.webhook_url,
+        'slack_channel': slack_details.slack_channel,
```
Comment (DavidLemayian) [inline ckanext/notify/actions.py:28]: I'm unsure but why do we need the slack_channel detail? Isn't `webhook_url` enough?

Comment (DavidLemayian) [reply ckanext/notify/actions.py:28]: @andela-iukwuoma ☝️

Comment (ignatiusukwuoma) [reply ckanext/notify/actions.py:28]: Actually, `webhook_url` is enough. We felt there was a need to display the `slack_channel` for the sake of the user who might not remember the `slack_channel` linked to the `webhook_url`.

Comment (ignatiusukwuoma) [reply ckanext/notify/actions.py:28]: @DavidLemayian

Comment (DavidLemayian) [reply ckanext/notify/actions.py:28]: @andela-iukwuoma Hmm.. As a "label"?

I'm wondering if the cost of this placebo outweighs the frustration of a user not knowing what format or whether it's the channel or account name etc. It's the first thing that I though about when I saw it.

I'd say we keep it for now and review user feedback / use.

**Pull request merged**